lib/_stream_writable.js

```diff
@@ -185,6 +187,12 @@ function WritableState(options, stream, isDuplex) {
   this.corkedRequestsFree = corkReq;
 }
 
+ObjectDefineProperty(WritableState.prototype, 'afterWriteTickInfo', {
+  enumerable: false,
```

Comment:
Nit: this is the default and can be omitted if you want.

Reply:
I’d prefer explicit over implicit.

Reply:
I don't have a strong opinion and fine either way.
I just remembered this PR https://github.com/nodejs/node/pull/31187 that actually removes explicit checks and thought we were moving in this direction and wanted to point this out. Though now, after checking the `stream_readable` and noticing that it still has all of the `enumerable: false` properties it doesn't look like it and the aforementioned PR is actually an inconsistency here :smile: .

Reply:
@lundibundi Thanks for the suggestion. Object property descriptor attributes are `false` by default because `undefined` is falsy when coerced to boolean. JavaScript engine searches accross the prototype chain and the first defined value (or `undefined` if the last null-prototype object has no such property defined) is casted to a boolean, and the search takes some time.

Performance is not critical here though (because this code runs only once), but in my understanding, omitting these properties is generally an anti-pattern.

Reply:
So are we deciding that we need an explicit `enumerable: false`? We had an earlier discussion that we can omit them. Since I was refactoring to group all these define property into a single block, writing up enumerable false for each property wasn't something that I had prefer. Thoughts?

Reply:
I find having the property explicitly defined can be confusing for some developers (it was for me in the beginning), i.e. it seem to mean that there is something special going on rather than being just a regular getter/setter.

No strong opinion though...

Reply:
Do we have a consensus here? I am not opposed to make it implicit if the PR cannot land otherwise, but after reading https://github.com/nodejs/node/pull/31187#discussion_r370340253 it sounds like the winds are blowing the opposite direction.

Note: This commit should also be backported to v12.x to actually fix https://github.com/nodejs/node/issues/31277.